File: aib/app/routes/intakes.py
"""
Intake REST endpoints — mirrors server/src/routes/intakes.js
"""

from fastapi import APIRouter, HTTPException

from app.services import extraction as extraction_svc
from app.services import session as session_svc

router = APIRouter(prefix="/api/intakes", tags=["intakes"])


@router.get("")
async def list_intakes():
    """List all completed intakes."""
    return extraction_svc.list_intakes()


@router.get("/{session_id}")
async def get_intake(session_id: str):
    """Get extracted intake data for a specific session."""
    intake = extraction_svc.get_intake(session_id)
    if not intake:
        raise HTTPException(status_code=404, detail="Intake not found")
    return intake


@router.get("/{session_id}/transcript")
async def get_transcript(session_id: str):
    """Get the full conversation transcript for a session."""
    session = session_svc.get_session_with_messages(session_id)
    if not session:
        raise HTTPException(status_code=404, detail="Session not found")

    transcript = [
        {
            "role": m["role"],
            "content": m["content"],
            "timestamp": m.get("created_at"),
        }
        for m in session.get("messages", [])
    ]

    return {
        "session_id": session["id"],
        "status": session["status"],
        "created_at": session.get("created_at"),
        "completed_at": session.get("completed_at"),
        "transcript": transcript,
    }

File: aib/server/src/routes/intakes.js
const express = require('express');
const router = express.Router();
const extractionService = require('../services/extraction');
const sessionService = require('../services/session');

// GET /api/intakes — List all completed intakes
router.get('/', async (req, res, next) => {
  try {
    const intakes = await extractionService.listIntakes();
    res.json(intakes);
  } catch (err) {
    next(err);
  }
});

// GET /api/intakes/:sessionId — Get extracted data for a specific intake
router.get('/:sessionId', async (req, res, next) => {
  try {
    const intake = await extractionService.getIntake(req.params.sessionId);
    if (!intake) {
      return res.status(404).json({ error: 'Intake not found' });
    }
    res.json(intake);
  } catch (err) {
    next(err);
  }
});

// GET /api/intakes/:sessionId/transcript — Get full conversation transcript
router.get('/:sessionId/transcript', async (req, res, next) => {
  try {
    const session = await sessionService.getSessionWithMessages(req.params.sessionId);
    if (!session) {
      return res.status(404).json({ error: 'Session not found' });
    }

    const transcript = session.messages.map(m => ({
      role: m.role,
      content: m.content,
      timestamp: m.created_at,
    }));

    res.json({
      session_id: session.id,
      status: session.status,
      created_at: session.created_at,
      completed_at: session.completed_at,
      transcript,
    });
  } catch (err) {
    next(err);
  }
});

module.exports = router;
